nixos/modules/services/networking/dnsmasq.md

```diff
@@ -0,0 +1,58 @@
+# Dnsmasq {#module-services-networking-dnsmasq}
+
+*Upstream documentation*: <https://dnsmasq.org/>
+
+Dnsmasq is integrated DNS, DHCP and TFTP server for small networks.
+
+## Configuration {#module-services-networking-dnsmasq-configuration}
+
+### An authoritative DHCP and DNS server on a home network {#module-services-networking-dnsmasq-configuration-home}
+
+On a home network, you can use Dnsmasq as a DHCP and DNS server. New devices on
+your network will be configured by Dnsmasq, and instructed to use it as the DNS
+server by default. This allows you to rely on your own server to perform DNS
+queries and caching, with DNSSEC enabled.
+
+The following example assumes that
+
+- you have disabled your router's integrated DHCP server, if it has one
+- your router's address is set in  [](#opt-networking.defaultGateway.address)
+- your system's Ethernet interface is `eth0`
+- you have configured the address(es) to forward DNS queries in [](#opt-networking.nameservers)
+
+```nix
+{
+  services.dnsmasq = {
+    enable = true;
+    settings = {
+      bind-interfaces = true;
+      bogus-priv = true;
+      dhcp-authoritative = true;
+      dhcp-host = [
+        # Give the current system a fixed address of 192.168.0.254
+        "dc:a6:32:0b:ea:b9,192.168.0.254,${config.networking.hostName},infinite"
+      ];
+      dhcp-option = [
+        "option:router,${config.networking.defaultGateway.address}"
+      ];
+      dhcp-range = [
+        # Range of IPv4 addresses to give out
+        # <range start>,<range end>,<lease time>
+        "192.168.0.10,192.168.0.253,24h"
+        # Enable stateless IPv6 allocation
+        "::f,::ff,constructor:eth0,ra-stateless"
+      ];
+      dhcp-rapid-commit = true;
+      dnssec = true;
+      domain-needed = true;
+      expand-hosts = true;
+      interface = "eth0";
+      local-service = true;
+      log-queries = true;
+      server = config.networking.nameservers;
+      # DNSSEC trust anchor. Source: https://data.iana.org/root-anchors/root-anchors.xml
+      trust-anchor = ".,20326,8,2,E06D44B80B8F1D39A95C0B0D7C65D08458E880409BBC683457104237C7F8EC8D";
+    };
+  };
+}
+```
```

Comment:
```suggestion
\```
## References

- Upstream website: https://dnsmasq.org/
- Manpage: https://dnsmasq.org/docs/dnsmasq-man.html 
- FAQ: https://dnsmasq.org/docs/FAQ
```

Reply:
The extra slash is because Github is bugged